coil-compose-base/src/main/java/coil/compose/Drawable.kt

```diff
@@ -0,0 +1,130 @@
+package coil.compose
+
+import android.graphics.drawable.Animatable
+import android.graphics.drawable.BitmapDrawable
+import android.graphics.drawable.ColorDrawable
+import android.graphics.drawable.Drawable
+import android.os.Build.VERSION.SDK_INT
+import android.os.Handler
+import android.os.Looper
+import android.view.View
+import androidx.compose.runtime.RememberObserver
+import androidx.compose.runtime.Stable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.graphics.asAndroidColorFilter
+import androidx.compose.ui.graphics.asImageBitmap
+import androidx.compose.ui.graphics.drawscope.DrawScope
+import androidx.compose.ui.graphics.drawscope.drawIntoCanvas
+import androidx.compose.ui.graphics.nativeCanvas
+import androidx.compose.ui.graphics.painter.BitmapPainter
+import androidx.compose.ui.graphics.painter.ColorPainter
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.graphics.withSave
+import androidx.compose.ui.unit.LayoutDirection
+import kotlin.LazyThreadSafetyMode.NONE
+import kotlin.math.roundToInt
+
+/**
+ * Convert this [Drawable] into a [Painter] using Compose primitives if possible.
+ */
+internal fun Drawable.toPainter(): Painter {
```

Comment:
I'll likely keep this in Accompanist as a new small lib. I'll follow up once I've refactored.

Reply:
https://github.com/google/accompanist/issues/540

Reply:
Sounds good :+1:. It's definitely super useful, but I didn't want to make it public API in Coil.

Reply:
Sounds good, it's definitely super useful, but I didn't want to make it public API in Coil. We'll swap to using Accompanist's dependency once it's published.